### PR TITLE
Get ready to launcharoo work history breaks!

### PR DESCRIPTION
**Pull request by chubberlisk**
## Context

Currently, we only show individual breaks when the `work_breaks` feature flag is on. However, to actually turn this on we have to consider the fact that we will turn on this feature while there are applications in progress i.e. there may be applications that have already used the "old" way of entering work history breaks (the capture-all text box). This means we must - Improvise. Adapt. Overcome!

## Changes proposed in this pull request

This PR does some smol but important things:

- updates the existing breaks component to show the break length in years and months (this was done for the placeholder before but I forgot about existing breaks) and does so by extracting a method into `ViewHelper` and reusing it in `BreakInWorkHistoryComponent`
- redirects to the review work history page if the `work_breaks` feature flag is off when attempting to go to a page to create, edit, etc. an individual break

And then, makes the new feature "backwards-compatible" by taking into consideration whether the feature flag is on and whether there is a value already for `work_history_breaks` (the attribute that stores the capture-all work history breaks in comparison to `application_work_history_breaks` an array of `ApplicationWorkHistoryBreak`s).

It also ensures that change and delete links don't show in Support.

## Guidance to review

- Have a look at `spec/components/work_history_review_component_spec.rb` does the specs make sense? Would highly recommend you also test through the UI make it easier to visualise.
- Does the redirect make sense?
- Is there anything I am missing?

## Link to Trello card

https://trello.com/c/jIWipybM/732-calculate-work-gaps

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by tvararu** [comment]
After (review app):

Screenshot 2020-02-19 at 13 19 02

File: spec/components/work_history_review_component_spec.rb [hunk 1]
```diff
@@ -97,39 +99,143 @@
...
+      end
+    end
+
+    context 'when the work breaks feature flag is on and work history breaks has a value' do
+      it 'renders component with capture all work breaks and without individual breaks' do
+        FeatureFlag.activate('work_breaks')
+        application_form.work_history_breaks = 'I was sleeping.'
+        application_form.application_work_history_breaks = []
+
+        result = render_inline(described_class.new(application_form: application_form))
```
Comment (tvararu) [inline spec/components/work_history_review_component_spec.rb]: You could create a helper method for this inside of this spec; allow passing in a few arguments which will let you reuse it for all the specs. Would be DRY-er, WDYT?

```ruby
result = render(feature_active: true, explanation: 'I was sleeping')

# ...

def render(feature_active: false, explanation: nil, breaks: [])
  if feature_active
    FeatureFlag.activate('work_breaks')
  else
    FeatureFlag.deactivate('work_breaks')
  end
  application_form.work_history_breaks = explanation
  application_form.application_work_history_breaks = breaks
 
  result = render_inline(described_class.new(application_form: application_form))
end
```

File: app/components/work_history_review_component.html.erb [hunk 2]
```diff
@@ -12,21 +12,19 @@
       <% end %>
     <% end %>
   <% elsif entry.is_a?(WorkHistoryWithBreaks::BreakPlaceholder) %>
-    <% if FeatureFlag.active?('work_breaks') %>
+    <% if FeatureFlag.active?('work_breaks') && @application_form.work_history_breaks.blank? && @editable %>
```
Comment (duncanjbrown) [inline app/components/work_history_review_component.html.erb]: Could this logic go into a named method on the component?

File: app/components/work_history_review_component.html.erb [hunk 3]
```diff
@@ -12,21 +12,19 @@
...
+    <%= render(BreakInWorkHistoryComponent.new(work_break: entry, editable: @editable)) %>
   <% end %>
 <% end %>
 
 <% if !show_missing_banner? && @application_form.application_work_experiences.empty? %>
   <%= render(SummaryCardComponent.new(rows: no_work_experience_rows, editable: @editable)) %>
 <% end %>
 
-<% if breaks_in_work_history? %>
+<% if breaks_in_work_history? && (@application_form.work_history_breaks || !FeatureFlag.active?('work_breaks')) %>
```
Comment (duncanjbrown) [inline app/components/work_history_review_component.html.erb]: And this one?

File: spec/components/work_history_review_component_spec.rb [hunk 4]
```diff
@@ -97,39 +99,143 @@
...
     end
 
-    context 'when work history breaks are editable' do
+    around do |example|
+      Timecop.freeze(Time.zone.local(2019, 10, 1)) do
+        example.run
+      end
+    end
+
+    context 'when capture-all work history breaks are editable' do
```
Comment (duncanjbrown) [inline spec/components/work_history_review_component_spec.rb]: Does this mean the "old" data? I'm not sure from the name

File: app/helpers/view_helper.rb [hunk 5]
```diff
@@ -77,6 +77,17 @@ def candidate_sign_in_url(candidate)
...
+  def format_months_to_years_and_months(number_of_months)
+    if number_of_months <= 12
+      pluralize(number_of_months, 'month')
+    else
+      years = number_of_months / 12
+      months = number_of_months % 12
+
+      "#{pluralize(years, 'year')} and #{pluralize(months, 'month')}"
+    end
+  end
```
Comment (duncanjbrown) [inline app/helpers/view_helper.rb]: How about `ActiveSupport::Duration`?

```
ActiveSupport::Duration.build(13.months).parts
=> {:years=>1, :months=>1}

 ActiveSupport::Duration.build(6.months).parts
=> {:months=>6}
```

File: spec/components/work_history_review_component_spec.rb (hunk 4 above)
Comment (chubberlisk) [reply spec/components/work_history_review_component_spec.rb]: Yep, trying to refer to this:

image

Apparently, "**_consolidated_** work history breaks" is the way to go according to Paul H and Emma. 😅 Do you think that clearer?

Comment (duncanjbrown) [reply spec/components/work_history_review_component_spec.rb]: Sounds good!

**Pull request merged**